src/cc2olx/models.py

```diff
@@ -329,115 +354,100 @@ def get_course_run(self):
         return 'run'
 
     def _extract(self):
-        settings = collect_settings()
-        workspace = settings['workspace']
-        path_extracted = filesystem.unzip_directory(self.file_path, workspace)
+        path_extracted = filesystem.unzip_directory(self.file_path, self.workspace)
         self.directory = path_extracted
-        manifest = os.path.join(path_extracted, MANIFEST)
+        manifest = path_extracted / MANIFEST
         return manifest
 
     def _update_namespaces(self, root):
-        ns = re.match('\{(.*)\}', root.tag).group(1)
-        version = re.match('.*/(imsccv\dp\d)/', ns).group(1)
+        ns = re.match('\{(.*)\}', root.tag).group(1)  # noqa: W605
+        version = re.match('.*/(imsccv\dp\d)/', ns).group(1)  # noqa: W605
```

Comment:
Maybe we want to listen to this error [1]? It looks like it will eventually be a `SyntaxError`.

- [1] https://www.flake8rules.com/rules/W605.html

Reply:
Well, since python expands `\d` and `\{` as is, I just changed these strings to be raw strings.